### PR TITLE
Update AutoScaling Blog to 24.07

### PR DESCRIPTION
**Pull request by indrajit96**
Update container file to point to 24.07 instead of 24.04
Update triton cli to point to 24.07 conversion scripts

### CODE REVIEW

**Review by whoisj — DISMISSED** [review]
LGTM :shipit:

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: efs-autoscaling-sc
+provisioner: efs.csi.aws.com
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml:5]: why are we checking in AWS specific solutions?

do we plan to provide GCloud and/or Azure options as well?

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/templates/deployment.yaml [hunk 2]
```diff
@@ -286,11 +283,14 @@ spec:
       - name: engine-repository
         hostPath:
           path: {{ printf "%s/models/%s/%dx%d/engines" $hostRootPath $model_lower (int $model_pp) (int $model_tp) }}
-          type: DirectoryOrCreate
+          type: DirectoryOrCreate
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/templates/deployment.yaml]: extra, errant whitespace at end of line.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/claim.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: efs-claim-autoscaling-2
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/claim.yaml:4]: we should pick a more meaningful, and less AWS specific name for this.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/pv.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: efs-autoscaling-pv-2
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/pv.yaml:4]: we should pick a more meaningful, and less AWS specific name for this.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/setup_ssh-efs.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: setup-ssh-efs
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/setup_ssh-efs.yaml:4]: we should pick a more meaningful, and less AWS specific name for this.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/containers/server.py [hunk 6]
```diff
@@ -212,66 +212,6 @@ def execute_triton(args):
     exit(result)
 
 
-# ---
-
-
-def initialize_model(args):
```
Comment (whoisj) [inline Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/containers/server.py:218]: with `initialize_model` removed, how is the model getting generated?

**Review by whoisj — CHANGES_REQUESTED** [review]

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml (hunk 1 above)
Comment (indrajit96) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml:5]: My ideal behind this was to show "some" kind of NFS solution.
AWS just was the most easiest to implement.
I have tried making it clear in our README that this is just ONE of the ways of doing it.
My thoughts were if we just say "use NFS" people might run into issues, hence better to show one working example.
I am happy to remove this if we don't want to check-in AWS specific solution.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/containers/server.py (hunk 6 above)
Comment (indrajit96) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/containers/server.py:218]: The model generation is now the responsibilty of the user and should be done before they start the service.
Steps for the same are outlined here https://github.com/triton-inference-server/tutorials/tree/ibhosale_update/Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing#model-preparation-steps

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml (hunk 1 above)
Comment (whoisj) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml:5]: then it should separate from the core charts. Putting AWS specific solutions into the core charts prevents users on other solutions from leveraging the charts completely.

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/pv.yaml (hunk 4 above)
Comment (indrajit96) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/pv.yaml]: Moved all pvc charts for aws under pvc_aws folder

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml (hunk 1 above)
Comment (indrajit96) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/storageclass.yaml]: Moved all pvc charts for aws under pvc_aws folder

File: Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/claim.yaml (hunk 3 above)
Comment (indrajit96) [reply Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/pvc/claim.yaml]: Renamed the claim form "efs-claim-autoscaling-2"
To
"nfs-claim-autoscaling-2" for to keep deployment.yaml AWS agnostic

**Review by whoisj — APPROVED** [review]

**Pull request merged**